### PR TITLE
android/ui: cleanup unused code

### PR DESCRIPTION
**Pull request by barnstar**
Updates tailscale/corp#18202

Removes unused code from various legacy ported classes.  Fixes a number of soft warnings.

Removes/renames a few composables we were no longer using.

### CODE REVIEW

File: android/src/main/java/com/tailscale/ipn/App.kt [hunk 1]
```diff
@@ -62,22 +48,17 @@ class App : Application(), libtailscale.AppContext {
...
+
     private val networkConnectivityRequest =
         NetworkRequest.Builder()
             .addCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)
             .addCapability(NetworkCapabilities.NET_CAPABILITY_NOT_VPN)
             .build()
-    lateinit var appInstance: App
 
-    @JvmStatic
-    fun startActivityForResult(act: Activity, intent: Intent?, request: Int) {
```
Comment (kari-ts) [inline android/src/main/java/com/tailscale/ipn/App.kt:77]: I think we still need this. We could definitely refactor this, but right now this launches Peer with the request VPN intent

File: android/src/main/java/com/tailscale/ipn/App.kt [hunk 2]
```diff
@@ -265,18 +243,6 @@ class App : Application(), libtailscale.AppContext {
     return null
   }
 
-  // attachPeer adds a Peer fragment for tracking the Activity
-  // lifecycle.
-  fun attachPeer(act: Activity) {
```
Comment (kari-ts) [inline android/src/main/java/com/tailscale/ipn/App.kt:270]: aha. I see, requestVpnPermission calls VpnService.prepare, so when App.prepareVpn calls VpnService.prepare, this is almost always null (vpn has been prepared), so we never need to call startActivityForResult right now. but if we're planning on moving requestVpnPermission() out of onStart, then we would still need this (and to actually call this function too)

Comment (kari-ts) [reply android/src/main/java/com/tailscale/ipn/App.kt:270]: we should probably refactor the vpn logic in general, it's overly confusing as is

**Comment by barnstar** [comment]
This has gone completely stale... I will redo.

**Pull request closed without merging**